### PR TITLE
fix: split connection pool based on credential

### PR DESCRIPTION
**Pull request by GaoleMeng**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java [hunk 1]
```diff
@@ -273,14 +278,18 @@ private StreamWriter(Builder builder) throws IOException {
         }
       }
       this.location = location;
+      CredentialsProvider credentialsProvider = client.getSettings().getCredentialsProvider();
+      log.warning("The credentials gotten is " + credentialsProvider.getCredentials());
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java:282]: Why are we printing this out?

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java [hunk 2]
```diff
@@ -273,14 +278,18 @@ private StreamWriter(Builder builder) throws IOException {
...
       // The first StreamWriter for a new stub will create the pool for the other
       // streams in the same region, meaning the per StreamWriter settings are no
       // longer working unless all streams share the same set of settings
       this.singleConnectionOrConnectionPool =
           SingleConnectionOrConnectionPool.ofConnectionPool(
               connectionPoolMap.computeIfAbsent(
-                  ConnectionPoolKey.create(location),
+                  ConnectionPoolKey.create(
+                      location,
+                      credentialsProvider != null ? credentialsProvider.getCredentials() : null),
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java:292]: Should we use a hash for the key instead? I am wondering if there could be a security concern here.

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java (hunk 1 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java]: wrong debug log, removed

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java (hunk 2 above)
Comment (GaoleMeng) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java:291]: Changed to use hash

**Review by yirutang — APPROVED** [review]

**Pull request merged**